CMakeLists.txt

```diff
@@ -73,6 +73,28 @@ add_test (NAME qrack_accuracy
     COMMAND accuracy
     )
 
+add_executable (grovers
```

Comment:
Let's make a separate cmake file for the examples that's included in this one.

Reply:
Consider in that example cmake file creating a macro https://cmake.org/cmake/help/v3.0/command/macro.html and using that to declare a bunch of these examples, as well as how to test them.

Reply:
We can discuss this further, but, specifically for examples, I like the idea of just being able to compile and link a single *.cpp file that produces an executable. It doesn't require any RTFM. It's immediately obvious what that is, how to build it, and how to run it, as kind of a "Hello World," without command line frustration.

Reply:
I can move this into a separate CMake file, though.